File: backend/src/main/kotlin/com/pp/trab3/model/Medicine.kt
package com.pp.trab3.model

import javax.persistence.ElementCollection
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.Id

@Entity
data class Medicine(@Id @GeneratedValue var id: Long?,
                    var name: String?,
                    @ElementCollection var functions: List<String>?)

File: frontend/record_doctor/lib/models/medicine.dart
class Medicine {
  int id;
  String name;
  List<String> functions;

  Medicine({this.id, this.name, this.functions});

  Medicine.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    name = json['name'];
    functions = json['functions'].cast<String>();
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['name'] = this.name;
    data['functions'] = this.functions;
    return data;
  }
}
